packages/wrangler/src/__tests__/index.test.ts

```diff
@@ -178,6 +180,144 @@ describe("wrangler", () => {
+          result: true,
+        },
+        {
+          text: "Would you like to use typescript?",
+          result: false,
+        }
+      );
+      await w("init");
+      expect(fs.existsSync("./package.json")).toBe(true);
+      const packageJson = JSON.parse(
+        fs.readFileSync("./package.json", "utf-8")
+      );
+      expect(packageJson.name).toEqual("worker"); // TODO: should we infer the name from the directory?
+      expect(packageJson.version).toEqual("0.0.1");
+      expect(fs.existsSync("./tsconfig.json")).toBe(false);
+    });
+
+    it("should not touch an existing package.json in the same directory", async () => {
+      mockConfirm({
+        text: "Would you like to use typescript?",
+        result: false,
+      });
+
+      fs.writeFileSync(
+        "./package.json",
+        JSON.stringify({ name: "test", version: "1.0.0" }),
+        "utf-8"
+      );
+
+      await w("init");
+      const packageJson = JSON.parse(
+        fs.readFileSync("./package.json", "utf-8")
+      );
+      expect(packageJson.name).toEqual("test");
+      expect(packageJson.version).toEqual("1.0.0");
+    });
+
+    it("should not touch an existing package.json in an ancestor directory", async () => {
+      mockConfirm({
+        text: "Would you like to use typescript?",
+        result: false,
+      });
+
+      fs.writeFileSync(
+        "./package.json",
+        JSON.stringify({ name: "test", version: "1.0.0" }),
+        "utf-8"
+      );
+
+      fs.mkdirSync("./sub-1/sub-2", { recursive: true });
+      process.chdir("./sub-1/sub-2");
+
+      await w("init");
+      expect(fs.existsSync("./package.json")).toBe(false);
+      expect(fs.existsSync("../../package.json")).toBe(true);
+
+      const packageJson = JSON.parse(
+        fs.readFileSync("../../package.json", "utf-8")
+      );
+      expect(packageJson.name).toEqual("test");
+      expect(packageJson.version).toEqual("1.0.0");
+    });
+
+    it("should create a tsconfig.json and install `workers-types` if none is found and user confirms", async () => {
+      mockConfirm(
+        {
+          text: "No package.json found. Would you like to create one?",
+          result: true,
+        },
+        {
+          text: "Would you like to use typescript?",
+          result: true,
+        }
+      );
+      await w("init");
+      expect(fs.existsSync("./tsconfig.json")).toBe(true);
+      const tsconfigJson = JSON.parse(
+        fs.readFileSync("./tsconfig.json", "utf-8")
+      );
+      expect(tsconfigJson.compilerOptions.types).toEqual([
+        "@cloudflare/workers-types",
+      ]);
+      const packageJson = JSON.parse(
+        fs.readFileSync("./package.json", "utf-8")
+      );
+      expect(packageJson.devDependencies).toEqual({
+        "@cloudflare/workers-types": expect.any(String),
+      });
+    });
+
+    it("should not touch an existing tsconfig.json in the same directory", async () => {
```

Comment:
maybe we could explicitly ask whether to install `@cloudflare/workers-types`, even if they have a tsconfig? In a future PR

Reply:
Yeah - I think there are a number of subtle aspects to this. For example

- whether we need to tweak the compilerOptions in a current tsconfig.json to match what we require. 
- whether to only touch tsconfig.json in the current working directory, since changing a higher up one might break other projects in a monorepo
- whether to check if the correct version of TS is installed
- whether to create a new tsconfig.json in the current directory even if there is a higher up one, which we then extend.